WebPaymentCoordinatorProxy only needs to run in the Networking process on iOS

gavin-apple opened the pull request:
#### 0d30a668978ba1bf095850644507a1ab24317bfd
<pre>
WebPaymentCoordinatorProxy only needs to run in the Networking process on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=288472">https://bugs.webkit.org/show_bug.cgi?id=288472</a>
<a href="https://rdar.apple.com/141212425">rdar://141212425</a>

Reviewed by Abrar Rahman Protyasha.

When APPLE_PAY_REMOTE_UI is enabled, WebPaymentCoordinatorProxy doesn&apos;t need to be brought up in the
UI process.

* Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.messages.in:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didAttachToRunningProcess):

Canonical link: <a href="https://commits.webkit.org/291313@main">https://commits.webkit.org/291313@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/456bbfda664c03ba661bef9709770324cea4faca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91836 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11366 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1462 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96789 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42467 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11928 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19852 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70490 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27991 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94837 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9084 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83623 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50817 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8812 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41677 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79130 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/1224 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98809 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18981 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/14011 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79519 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19232 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79506 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78741 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19636 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23277 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/901 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/12031 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18962 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24184 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18663 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22121 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20419 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.messages.in:
```diff
@@ -24,6 +24,7 @@
 
+// Currently DispatchedTo=Networking when APPLE_PAY_REMOTE_UI and DispatchedTo=UI when !APPLE_PAY_REMOTE_UI
 [
     DispatchedFrom=WebContent,
```
Is it not possible to express something like this? Would be handy to bake this into the specification rather than through a comment!

```
[
    DispatchedFrom=WebContent,
#if ENABLE(APPLE_PAY_REMOTE_UI)
    DispatchedTo=Networking,
#else
    DispatchedTo=UI,
#endif
    EnabledBy=ApplePayEnabled
]
```

aprotyas commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -1559,7 +1559,7 @@ void WebPageProxy::didAttachToRunningProcess()
 #endif
 
-#if ENABLE(APPLE_PAY)
+#if ENABLE(APPLE_PAY) && !ENABLE(APPLE_PAY_REMOTE_UI)
```
LGTM, we probably want a better name for this now, since APPLE_PAY_REMOTE_UI is not too telling...

aprotyas replied:
... out of scope for this patch, of course

gavin-apple committed:
WebPaymentCoordinatorProxy only needs to run in the Networking process on iOS
https://bugs.webkit.org/show_bug.cgi?id=288472
rdar://141212425

Reviewed by Abrar Rahman Protyasha.

When APPLE_PAY_REMOTE_UI is enabled, WebPaymentCoordinatorProxy doesn't need to be brought up in the
UI process.

* Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.messages.in:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didAttachToRunningProcess):

Canonical link: https://commits.webkit.org/291313@main
